src/os/GenericObjectMap.cc

```diff
@@ -1099,7 +1099,14 @@ int GenericObjectMap::list_objects(const coll_t &cid, ghobject_t start, int max,
       break;
     }
 
+    if (header.oid >= end) {
+      if (next)
+	*next = ghobject_t::get_max();
```

Comment:
It looks a little confuse to me, is "end" must be "ghobject_t::get_max()"? 
It should be " *next = header.oid;" more reasonable?